src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.cs

```diff
@@ -271,9 +271,9 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
         // It also provides a human-readable string to explain the reason. It will be emitted by the source generator
         // as a comment into the C# code, hence there's no need to localize.
         /// </remarks>
-        private static bool SupportsCodeGeneration(RegexMethod method, [NotNullWhen(false)] out string? reason)
+        private static bool SupportsCodeGeneration(LanguageVersion languageVersion, [NotNullWhen(false)] out string? reason)
         {
-            if (method.MethodSyntax.SyntaxTree.Options is CSharpParseOptions { LanguageVersion: <= LanguageVersion.CSharp10 })
```

Comment:
Ideally, `MethodSyntax` shouldn't exist in the `RegexMethod` model at all. I could eliminate this usage, but not the other one. (the other usage can be changed to be `Location` instead of `MethodDeclarationSyntax` but that is not good as well)

So I simplified this usage for now.

Someone from Roslyn team should be able to advise about the other usage.